symengine/polys.h

```diff
@@ -0,0 +1,70 @@
+/**
+ *  \file polys.h
+ *  Class for sparse Polynomial: Polynomial
+ *
+ **/
+#ifndef SYMENGINE_POLYS_H
+#define SYMENGINE_POLYS_H
+
+#include <symengine/basic.h>
+#include <symengine/dict.h>
+
+namespace SymEngine {
+//! Polynomial Class
+class Polynomial : public Basic{
+public:
+    //! `vars_` : Variables of the Polynomial
+    //! `polys_set_` : holds the Polynomial
+    vec_symbol vars_;
+    hash_set polys_set_;
```

Comment:
@Sumith1896 for the forward declaration I can add a ``hash_set_fwd.hpp" header in Piranha, so you can include that and make sure that the ``hash_set`` declaration is always in sync with Piranha's.